File: budget-buddy-front/src/app/entity/CategoricalBudgetOverview.ts
import {Category} from "../admin/category/category";
import {Transaction} from "../admin/transaction/transaction";
import {BudgetPerMonthPerCategory} from "./BudgetPerMonthPerCategory";

export interface CategoricalBudgetOverview {
  category: Category;
  budgetsPerMonth: BudgetPerMonthPerCategory[];
  transactions: Transaction[];
  total: number;
}

File: budget-buddy-back/src/main/java/be/yorian/budgetbuddy/dto/category/CategoricalBudgetOverview.java
package be.yorian.budgetbuddy.dto.category;

import be.yorian.budgetbuddy.entity.Category;

import java.util.List;

public record CategoricalBudgetOverview(
        Category category,
        List<BudgetPerMonthPerCategory> budgetsPerMonth) {
}
